src/java/map/FloatMask.java

```diff
@@ -306,6 +324,58 @@ public FloatMask clampMax(float val) {
         return this;
     }
 
+    public FloatMask enlarge(int size) {
+        float[][] largeMask = new float[size][size];
+        int smallX;
+        int smallY;
+        for (int x = 0; x < size; x++) {
+            smallX = StrictMath.min(x / (size / getSize()), getSize() - 1);
+            for (int y = 0; y < size; y++) {
+                smallY = StrictMath.min(y / (size / getSize()), getSize() - 1);
+                largeMask[x][y] = get(smallX, smallY);
+            }
+        }
+        mask = largeMask;
+        applySymmetry(symmetryHierarchy.getSpawnSymmetry());
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask enlarge2 (int size) {
+        float[][] largeMask = new float[size][size];
+        int oldSize = getSize();
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                largeMask[x][y] = get(Math.round(x / (size / oldSize)), Math.round(y / (size / oldSize)));
+            }
+        }
+        mask = largeMask;
+        applySymmetry(symmetryHierarchy.getSpawnSymmetry());
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask shrink2 (int size) {
+        float[][] smallMask = new float[size][size];
+        int oldSize = getSize();
+        float sum = 0;
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                for (int z = 0; z < (oldSize / size); z++) {
+                    for (int w = 0; w < (oldSize / size); w++) {
+                        sum += get((x * oldSize / size) + z, (y * oldSize / size) + w);
+                    }
+                }
+                smallMask[x][y] = sum / oldSize * size;
+                sum = 0;
+            }
+        }
+        mask = smallMask;
+        applySymmetry(symmetryHierarchy.getSpawnSymmetry());
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
```

Comment:
This is very inefficient as it recalculates the average for every pixel. Instead you could smooth the mask and then shrink and this would be equivalent and more efficient.